### PR TITLE
Refresh

### PR DESCRIPTION
**Pull request by rieschl**
* require PHP 7.4
* update code quality tools
* import global functions
* fix some minor bugs in doctypes
* add GitHub CI

### Todo

- [x] deprecate `TranslationMapFactoryInterface` and `TranslationMapFactory`. Implement a named constructor `public static function create(array $mapData);` directly on `TranslationMap`
- [x] deprecate `TranslationExtractorInterface` and `TranslationExtractor`. Implement a method `TranslationMap::pick(LanguagePriority $languages)` instead.

### CODE REVIEW

File: src/Language/Language.php [hunk 1]
```diff
@@ -6,14 +6,17 @@
...
 use Eventjet\I18n\Exception\InvalidLanguageFormatException;
 
+use function preg_match;
+use function sprintf;
+use function strpos;
+use function substr;
+
 class Language implements LanguageInterface
 {
     /** @var Language[] */
```
Comment (MidnightDesign) [inline src/Language/Language.php]:
```suggestion
    /** @var array<string, Language> */
```

File: src/Language/LanguagePriority.php [hunk 2]
```diff
@@ -7,11 +7,16 @@
...
 use function assert;
+use function count;
+use function current;
+use function key;
+use function next;
+use function reset;
 
 class LanguagePriority implements LanguagePriorityInterface
 {
     /** @var array<int, LanguageInterface> */
```
Comment (MidnightDesign) [inline src/Language/LanguagePriority.php]:
```suggestion
    /** @var list<LanguageInterface> */
```

**Comment by rieschl** [comment]
@MidnightDesign @wolfgangschaefer  should we also deprecate https://github.com/eventjet/eventjet-i18n/blob/f1a2ab2859f631bb8bb4a39509fd84049dc77a45/src/Translate/Factory/TranslationMapFactoryInterface.php#L9
and https://github.com/eventjet/eventjet-i18n/blob/f1a2ab2859f631bb8bb4a39509fd84049dc77a45/src/Translate/TranslationExtractorInterface.php#L10
?

**Review by wolfgangschaefer — APPROVED** [review]

File: psalm.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        <directory name="src"/>
+        <directory name="tests"/>
+        <ignoreFiles>
+            <directory name="vendor"/>
+        </ignoreFiles>
+    </projectFiles>
+
+    <issueHandlers>
+        <DeprecatedClass errorLevel="suppress"/>
+        <DeprecatedInterface errorLevel="suppress"/>
```
Comment (MidnightDesign) [inline psalm.xml]:
```suggestion
        <DeprecatedClass errorLevel="info"/>
        <DeprecatedInterface errorLevel="info"/>
```

File: src/Translate/TranslationMap.php [hunk 4]
```diff
@@ -30,19 +39,65 @@ public function __construct(array $translations)
...
+    }
+
+    private static function getExtractor(): TranslationExtractor
+    {
+        if (self::$extractor !== null) {
+            return self::$extractor;
+        }
+        $extractor = new TranslationExtractor();
+        self::$extractor = $extractor;
+        return $extractor;
```
Comment (MidnightDesign) [inline src/Translate/TranslationMap.php]:
```suggestion
        if (self::$extractor === null) {
            self::$extractor = new TranslationExtractor();
        }
        return self::$extractor;
```

**Review by MidnightDesign — APPROVED** [review]

**Pull request merged**